internal/pkg/docker/dockerengine/dockerengine.go

```diff
@@ -133,10 +140,13 @@ func (c CmdClient) Build(in *BuildArguments) error {
 
 // Login will run a `docker login` command against the Service repository URI with the input uri and auth data.
 func (c CmdClient) Login(uri, username, password string) error {
-	err := c.runner.Run("docker",
-		[]string{"login", "-u", username, "--password-stdin", uri},
-		exec.Stdin(strings.NewReader(password)))
-
+	var err error
+	loginOnce.Do(
+		func() {
+			err = c.runner.Run("docker",
+				[]string{"login", "-u", username, "--password-stdin", uri},
+				exec.Stdin(strings.NewReader(password)))
+		})
```

Comment:
Instead of a global `sync.Once` to me this is hinting that we need a different constructor that enables parallel build and push for example:

```go
func NewWithLogin(exec Cmd, registryURI string, username, passwd string) (*CmdClient, error) {
   client := New(exec)
   if err := client.Login(registryURI, username, passwd); err != nil { return nil, err }
   return client, nil
}
```

And then all the clients can safely use this client. Would that work? 💭 


Or alternatively, we return a specific error here that catches "the specified item already exists" if the client receives that error they ignore it on purpose 💭